src/components/AccountDropdown/AccountDropdown.tsx

```diff
@@ -39,13 +39,18 @@ import { AccountChildOption } from './AccountChildOption'
 import { AccountSegment } from './AccountSegement'
 
 type AccountDropdownProps = {
+  accountId?: AccountId // if provided - will force this component into this state
   assetId: AssetId
```

Comment:
We probably want `assetId` and `accountId` to be mutually exclusive.

If we pass an `accountId` the `assetId` has no effect, and is meaningless.
This is confusing as a consumer of this component.

Reply:
this is actually a really good point. discriminated union props?

Reply:
Yup, that's what I had in mind.

```ts
type AccountDropdownProps =
  | {
      buttonProps?: ButtonProps
      onChange: (accountId: AccountId) => void
    } & ({ assetId: AssetId } | { accountId: AccountId })
```

Or something.